### ISSUE TITLE
Again ORA-38104 on newer versions of Oracle

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#10056]
The workaround introduced with #8158 for ORA-38104 errors no longer works. We've used row value expressions like this to "outsmart" Oracle's parser:

```sql
merge into "TEST"."T_BOOK"
using (
  select 1 "one"
  from dual
)
on (("TEST"."T_BOOK"."ID", 1) = ((5, 1))) -- Trick here
when matched then update set
  "TEST"."T_BOOK"."ID" = 5,
  "TEST"."T_BOOK"."AUTHOR_ID" = 1,
  "TEST"."T_BOOK"."TITLE" = 'XX',
  "TEST"."T_BOOK"."PUBLISHED_IN" = 2000,
  "TEST"."T_BOOK"."LANGUAGE_ID" = 1,
  "TEST"."T_BOOK"."REC_VERSION" = 1,
  "TEST"."T_BOOK"."REC_TIMESTAMP" = timestamp '2020-04-09 14:26:25.329' where "TEST"."T_BOOK"."ID" = 5
when not matched then insert (
  "ID",
  "AUTHOR_ID",
  "TITLE",
  "PUBLISHED_IN",
  "LANGUAGE_ID",
  "REC_VERSION",
  "REC_TIMESTAMP"
)
values (
  5, 
  1, 
  'XX', 
  2000, 
  1, 
  1, 
  timestamp '2020-04-09 14:26:25.329'
)
```

If instead we add another dummy predicate, things seem to work again:

```sql
-- This works (combining both tricks)
on (("TEST"."T_BOOK"."ID", 1) = ((5, 1)) OR (1 = 0))

-- This doesn't work
on ("TEST"."T_BOOK"."ID" = 5 OR (1 = 0))
```

This is really ridiculous, but here we are. The new plan is the same as the old one. Seems we're outsmarting only the parser, not the optimiser.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.14.0. Backport scheduled for 3.13.2 (#10057) and 3.12.5 (#10058)

**Comment by lukaseder** [comment]
These workarounds will be superseded by https://github.com/jOOQ/jOOQ/issues/11205
